satellite/gracefulexit/endpoint_test.go

```diff
@@ -493,6 +495,126 @@ func TestSuccessPointerUpdate(t *testing.T) {
+				Timestamp: header.GetCreationTime(),
+				Signature: header.GetSignature(),
+			}
+
+			newPieceHash := &pb.PieceHash{
+				PieceId:   m.TransferPiece.AddressedOrderLimit.Limit.PieceId,
+				Hash:      originalPieceHash.Hash,
+				PieceSize: originalPieceHash.PieceSize,
+				Timestamp: time.Now(),
+			}
+
+			receivingIdentity := nodeFullIDs[m.TransferPiece.AddressedOrderLimit.Limit.StorageNodeId]
+			require.NotNil(t, receivingIdentity)
+
+			// get the receiving node piece count before processing
+			recNodeID = receivingIdentity.ID
+
+			// update pointer to include the new receiving node before responding to satellite
+			keys, err := satellite.Metainfo.Database.List(ctx, nil, 1)
+			require.NoError(t, err)
+			path := string(keys[0])
+			pointer, err := satellite.Metainfo.Service.Get(ctx, path)
+			require.NoError(t, err)
+			require.NotNil(t, pointer.GetRemote())
+			require.True(t, len(pointer.GetRemote().GetRemotePieces()) > 0)
+
+			pieceToRemove := make([]*pb.RemotePiece, 1)
+			pieceToAdd := make([]*pb.RemotePiece, 1)
+			pieces := pointer.GetRemote().GetRemotePieces()
+
+			for _, piece := range pieces {
+				if pieceToRemove[0] == nil && piece.NodeId != exitingNode.ID() {
+					pieceToRemove[0] = piece
+					continue
+				}
+			}
+
+			// create a piece with deleted piece number and receiving node ID from the pointer
+			pieceToAdd[0] = &pb.RemotePiece{
+				PieceNum: pieceToRemove[0].PieceNum,
+				NodeId:   recNodeID,
+			}
+
+			_, err = satellite.Metainfo.Service.UpdatePieces(ctx, path, pointer, pieceToAdd, pieceToRemove)
+			require.NoError(t, err)
+
+			signer := signing.SignerFromFullIdentity(receivingIdentity)
+
+			signedNewPieceHash, err := signing.SignPieceHash(ctx, signer, newPieceHash)
+			require.NoError(t, err)
+
+			success := &pb.StorageNodeMessage{
+				Message: &pb.StorageNodeMessage_Succeeded{
+					Succeeded: &pb.TransferSucceeded{
+						OriginalPieceId:      m.TransferPiece.OriginalPieceId,
+						OriginalPieceHash:    originalPieceHash,
+						OriginalOrderLimit:   &orderLimit,
+						ReplacementPieceHash: signedNewPieceHash,
+					},
+				},
+			}
+			err = processClient.Send(success)
+			require.NoError(t, err)
+		default:
+			t.FailNow()
+		}
+
+		_, err = processClient.Recv()
+		require.Error(t, err)
+		require.True(t, errs2.IsRPC(err, rpcstatus.Internal))
+
+		// check exiting node is still in the pointer
+		keys, err := satellite.Metainfo.Database.List(ctx, nil, 1)
+		require.NoError(t, err)
+		path := string(keys[0])
+		pointer, err := satellite.Metainfo.Service.Get(ctx, path)
+		require.NoError(t, err)
+		require.NotNil(t, pointer.GetRemote())
+		require.True(t, len(pointer.GetRemote().GetRemotePieces()) > 0)
+
+		pieces := pointer.GetRemote().GetRemotePieces()
+
+		pieceMap := make(map[storj.NodeID]int)
+		for _, piece := range pieces {
+			pieceMap[piece.NodeId]++
+		}
+
+		exitingNodeID := exitingNode.ID()
+		count, ok := pieceMap[exitingNodeID]
+		require.True(t, ok)
```

Comment:
nit - there is probably no need to check `ok` since `count` will be 0 otherwise.